fl4health/clients/basic_client.py

```diff
@@ -178,11 +178,13 @@ def train_by_epochs(
         for local_epoch in range(epochs):
             self.train_metric_meter.clear()
             self.train_loss_meter.clear()
-            for input, target in self.train_loader:
+            for step, (input, target) in enumerate(self.train_loader):
                 input, target = input.to(self.device), target.to(self.device)
                 losses, preds = self.train_step(input, target)
                 self.train_loss_meter.update(losses)
                 self.train_metric_meter.update(preds, target)
+                actual_step = int(local_epoch * len(self.train_loader) + step)
```

Comment:
I don't think this calculation is wrong or anything, but, rather than performing this calculation each time, would it not be simpler to implement a step counter that is, above line 178 do `local_steps=0`
and inside the for loop just do `local_steps+=1` and send local_steps to `self.update_after_step(local_steps)`

Reply:
Updated in [349c7cb](https://github.com/VectorInstitute/FL4Health/pull/62/commits/349c7cb7ba64abfdb91b1e5c77eca6d7ac30c789)